src/mlpack/methods/hoeffding_trees/hoeffding_tree_impl.hpp

```diff
@@ -804,7 +804,7 @@ void HoeffdingTree<
   ar(CEREAL_POINTER(dimensionMappings));
 
   // Special handling for const object.
-  ar(CEREAL_POINTER(const_cast<data::DatasetInfo*>(datasetInfo)));
+  ar(CEREAL_POINTER(const_cast<data::DatasetInfo*&>(datasetInfo)));
```

Comment:
Did this somehow break the `HoeffdingTree` serialization test on OS X?  If so I guess we should revert the suggestion. :'(  It did seem to work okay before.

Reply:
ok, I will bring back the old temporary pointer for this one.